internal/adapters/events/events_forwarder.go

```diff
@@ -217,12 +219,30 @@ func fromPlayerEventTypeToGrpcPlayerEventType(eventType events.PlayerEventType)
 	}
 }
 
-func handlerGrpcClientResponse(forwarder entities.Forwarder, eventResponse *pb.Response, err error) error {
+// handlerGrpcClientResponse checks if the gRPC response had unexpected communication or network errors.
+func handlerGrpcClientResponse(forwarder entities.Forwarder, eventResponse *pb.Response, err error) (codes.Code, error) {
 	if err != nil {
-		return err
+		grpcStatus, ok := status.FromError(err)
+		if !ok {
+			return codes.Unknown, errors.NewErrUnexpected("failed to forward event room at \"%s\" with unknown grpc code: %s", forwarder.Name, err.Error())
+		}
+
+		return grpcStatus.Code(), errors.NewErrUnexpected("failed to forward event room at \"%s\" with code %s", forwarder.Name, grpcStatus.Code().String())
 	}
-	if eventResponse.Code != 200 {
-		return errors.NewErrUnexpected("failed to forward event room at \"%s\"", forwarder.Name)
+
+	var ret codes.Code
+	// Matchmaker sends the status code as a http one, we normalize it to grpc codes for logs and metrics.
+	switch eventResponse.Code {
+	case 200:
+		ret = codes.OK
+	case 404:
+		ret = codes.NotFound
+	case 500:
+		ret = codes.Internal
+	default:
+		ret = codes.Unknown
```

Comment:
Add a case for `406` which MM returns when room does not exist in its Redis.  We can map to `codes.FailedPrecondition`

Reply:
This is something we expect to receive when forwarding events, we should add some documentation about this, thus whoever develops a forwarder will know what to return for specific use cases.

Reply:
Added both suggestions